File: Leetcode/Javascript/Array and Hashing/easy-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length !== t.length) {
    return false;
  }

  let sCount = {};
  let tCount = {};
  for (let i = 0; i < s.length; i++) {
    let ch1 = s.charAt(i);
    let ch2 = t.charAt(i);
    if (!sCount[ch1]) {
      sCount[ch1] = 1;
    } else {
      sCount[ch1] += 1;
    }
    if (!tCount[ch2]) {
      tCount[ch2] = 1;
    } else {
      tCount[ch2] += 1;
    }
  }

  for (let key in sCount) {
    if (sCount[key] !== tCount[key]) {
      return false;
    }
  }
  for (let key in tCount) {
    if (tCount[key] !== sCount[key]) {
      return false;
    }
  }
  return true;
};

File: Leetcode/Python/Array & Hashing/easy_valid_anagram.py
def isAnagram(s, t):
    s_count = {}
    t_count = {}

    for i in s:
        s_count[i] = (s_count.get(i, 0) + 1)
        
    for j in t:
        t_count[j] = (t_count.get(j, 0) + 1)
    
    return s_count == t_count

print(isAnagram("rat", "car"))